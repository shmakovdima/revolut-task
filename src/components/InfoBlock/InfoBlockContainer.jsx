import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import {
  getExchangeRateFetching,
  getExchangeRateError,
  getExchangeRateByOne,
} from '../../store/selectors/exchangeRate';

import { swapCurrencyAction } from '../../store/actions/userInputs';

import { getCurrencyFrom, getCurrencyTo } from '../../store/selectors/userInputs';

import { getCurrencyTypes } from '../../utils';

import InfoBlockView from './InfoBlockView';

const InfoBlock = (props) => {
  const [hovered, setHovered] = useState(false);

  return (
    <InfoBlockView
      {...props}
      hovered={hovered}
      setHovered={setHovered}
    />
  );
};

InfoBlock.propTypes = {
  costOne: PropTypes.string.isRequired,
  isFetching: PropTypes.bool.isRequired,
  isError: PropTypes.bool.isRequired,
  currencyFrom: PropTypes.oneOf(getCurrencyTypes).isRequired,
  currencyTo: PropTypes.oneOf(getCurrencyTypes).isRequired,
  swapCurrency: PropTypes.func.isRequired,
};


const mapStateToProps = state => ({
  isFetching: getExchangeRateFetching(state),
  isError: getExchangeRateError(state),
  costOne: getExchangeRateByOne(state),
  currencyFrom: getCurrencyFrom(state),
  currencyTo: getCurrencyTo(state),
});

const mapDispatchToProps = dispatch => ({
  swapCurrency: () => dispatch(swapCurrencyAction()),
});

export { InfoBlock };

export default connect(mapStateToProps, mapDispatchToProps)(React.memo(InfoBlock));
