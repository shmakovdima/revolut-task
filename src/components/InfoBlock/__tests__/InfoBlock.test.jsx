import React from 'react';
import { render } from 'enzyme';
import { InfoBlock } from '../InfoBlockContainer';

const props = {
  currencyFrom: 'GBP',
  currencyTo: 'USD',
  costOne: '34.4554',
  isFetching: false,
  isError: false,
  swapCurrency: jest.fn(),
};

describe('Component [InfoBlock]', () => {
  test('Basic render', () => {
    const component = render(<InfoBlock
      {...props}
    />);
    expect(component).toMatchSnapshot();
  });
});
