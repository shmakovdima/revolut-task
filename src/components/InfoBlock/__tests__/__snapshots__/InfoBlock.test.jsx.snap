// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component [InfoBlock] Basic render 1`] = `
<div
  class="sc-bdVaJa ftuCeF"
>
  <button
    class="sc-bwzfXH gHjLTq"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-exchange-alt fa-w-16 fa-lg "
      data-icon="exchange-alt"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 168v-16c0-13.255 10.745-24 24-24h360V80c0-21.367 25.899-32.042 40.971-16.971l80 80c9.372 9.373 9.372 24.569 0 33.941l-80 80C409.956 271.982 384 261.456 384 240v-48H24c-13.255 0-24-10.745-24-24zm488 152H128v-48c0-21.314-25.862-32.08-40.971-16.971l-80 80c-9.372 9.373-9.372 24.569 0 33.941l80 80C102.057 463.997 128 453.437 128 432v-48h360c13.255 0 24-10.745 24-24v-16c0-13.255-10.745-24-24-24z"
        fill="currentColor"
      />
    </svg>
  </button>
  <div
    class="sc-htpNat kddTOz"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-chart-line fa-w-16 fa-1x sc-ifAKCX fNIInB"
      data-icon="chart-line"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zM464 96H345.94c-21.38 0-32.09 25.85-16.97 40.97l32.4 32.4L288 242.75l-73.37-73.37c-12.5-12.5-32.76-12.5-45.25 0l-68.69 68.69c-6.25 6.25-6.25 16.38 0 22.63l22.62 22.62c6.25 6.25 16.38 6.25 22.63 0L192 237.25l73.37 73.37c12.5 12.5 32.76 12.5 45.25 0l96-96 32.4 32.4c15.12 15.12 40.97 4.41 40.97-16.97V112c.01-8.84-7.15-16-15.99-16z"
        fill="currentColor"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-pound-sign fa-w-10 fa-sm sc-bZQynM gEfSqh"
      data-icon="pound-sign"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M308 352h-45.495c-6.627 0-12 5.373-12 12v50.848H128V288h84c6.627 0 12-5.373 12-12v-40c0-6.627-5.373-12-12-12h-84v-63.556c0-32.266 24.562-57.086 61.792-57.086 23.658 0 45.878 11.505 57.652 18.849 5.151 3.213 11.888 2.051 15.688-2.685l28.493-35.513c4.233-5.276 3.279-13.005-2.119-17.081C273.124 54.56 236.576 32 187.931 32 106.026 32 48 84.742 48 157.961V224H20c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h28v128H12c-6.627 0-12 5.373-12 12v40c0 6.627 5.373 12 12 12h296c6.627 0 12-5.373 12-12V364c0-6.627-5.373-12-12-12z"
        fill="currentColor"
      />
    </svg>
    1 = 
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-dollar-sign fa-w-9 fa-sm sc-bZQynM gEfSqh"
      data-icon="dollar-sign"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 288 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M209.2 233.4l-108-31.6C88.7 198.2 80 186.5 80 173.5c0-16.3 13.2-29.5 29.5-29.5h66.3c12.2 0 24.2 3.7 34.2 10.5 6.1 4.1 14.3 3.1 19.5-2l34.8-34c7.1-6.9 6.1-18.4-1.8-24.5C238 74.8 207.4 64.1 176 64V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48h-2.5C45.8 64-5.4 118.7.5 183.6c4.2 46.1 39.4 83.6 83.8 96.6l102.5 30c12.5 3.7 21.2 15.3 21.2 28.3 0 16.3-13.2 29.5-29.5 29.5h-66.3C100 368 88 364.3 78 357.5c-6.1-4.1-14.3-3.1-19.5 2l-34.8 34c-7.1 6.9-6.1 18.4 1.8 24.5 24.5 19.2 55.1 29.9 86.5 30v48c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-48.2c46.6-.9 90.3-28.6 105.7-72.7 21.5-61.6-14.6-124.8-72.5-141.7z"
        fill="currentColor"
      />
    </svg>
    34.4554
  </div>
</div>
`;
